dune/iga/io/igadatacollector.hh

```diff
@@ -43,23 +47,19 @@ namespace Dune::Vtk {
       for (auto element : elements(gridView_)) {
         const size_t elementId = indexSet.index(element);
 
-        auto elementRepr       = element.impl().trimmedElementRepresentation();
-        auto subGridView       = elementRepr->gridView();
-        auto verticesInSubGrid = subGridView.size(dim);
+        auto verticesInSubGrid = geometries_.nVertices(elementId);
 
-        numCells_ += subGridView.size(0);
-        const auto& subGridIndexSet = subGridView.indexSet();
+        numCells_ += geometries_.nElements(elementId);
 
-        for (auto gt : subGridIndexSet.types(0))
-          pointSets_.try_emplace(gt, 1);
+        pointSets_.try_emplace(geometries_.geometryType(elementId), 1);
 
         for (auto& [type, pointSet] : pointSets_)
           if (pointSet.size() == 0) pointSet.build(type);
 
-        for (auto subGridVertex : vertices(subGridView)) {
-          std::size_t idx = subGridIndexSet.index(subGridVertex);
-          vertexIndex_.emplace(std::array<std::size_t, 2>({elementId, idx}), vertexCounter++);
-        }
+        std::size_t vIdx = 0;
+        for (auto& subGridVertex : geometries_.getVertices(elementId))
```

Comment:
This can be simplified to: `for (std::size_t vIdx = 0; auto& subGridVertex : geometries_.getVertices(elementId))`